src/log.h

```diff
@@ -86,7 +88,8 @@ class LogStream
       buf_ << v;
     return *this;
   }
-  ~LogStream();
+  virtual ~LogStream();
+  friend class LogStreamFatal;
```

Comment:
Friend classes are generally a code smell. Why not make `private` members `protected` instead?